README.md

```diff
@@ -1,5 +1,102 @@
 # kappa-core
 
+## **kappa-core WIP rewrite** ##
+
+kappa-core is a database abstraction for append-only logs and add-only sets. A kappa-core is a container for pairs of sources and views, called *flows*. In each flow, data flows from a source into a view. Sources have a pull function that fetches new messages since the last pull. Views have a map function which is called for each batch of messages from the source.
+
+kappa-core itself is dependencyless, but this module also contains stateful source handlers for [hypercores](https://github.com/mafintosh/hypercore), [multifeed](https://github.com/kappa-core/multifeed), and [corestore](https://github.com/andrewosh/corestore).
+
+## API
+
+`const { Kappa } = require('kappa-core')`
+
+#### `const kappa = new Kappa()`
+
+Create a new kappa core.
+
+#### `kappa.use(name, source, view)`
+
+Register a flow. 
+
+* `name` (string) the name of the flow, has to be unique per kappa core
+* `source` object with properties:
+  * `open: function (flow, next)`: (optional) Handler to call on open
+  * `pull: function (next)`: Pull new messages from the view. Should call `next` with either nothing or an object that looks like this:
+    ```javascript
+    {
+        messages: [], // array of messages
+        finished: true, // if set to false, signal that more messages are pending
+        onindexed: function (cb) {} // will be called when the view finished indexing
+    }
+    ```
+    The source has to track its state, so that subsequent calls to `pull()` do not return the same messages. Use the `onindexed` callback to update state.
+
+    There are several source handlers included in kappa-core (TODO: document sources). See the tests and sources directories.
+
+    A simple state handler that perists state in a leveldb (or in memory) is included and used by the bundled source handler. It's available for use by custom sources on `Kappa.SimpleState`. If the bundled source handlers get a `db` option passed with levelup instance, the source state will be persisted.
+
+  * `reset: function (cb)`: Reset internal state (next pull should start at the beginning). This handler is called by the kappa-core if the view's `version` property increases and thus wants to restart indexing.
+  * `storeVersion: function (version, cb)`: Store the flow version somewhere
+  * `fetchVersion: function (cb)`: Fetch the version stored with storeVersion
+* `view` object with properties 
+  * `map: function (msgs, next)` (required) 
+  * `open: function (flow, next)` (optional)
+  * `filter: function (msgs, next)` (optional)
+  * `version: int`
+
+Both `source` and `view` can have an `api` property with an object of function. The functions are exposed on `kappa.view[name]` / `kappa.source[name]`. Their `this` object refers to the flow, and their first parameter is the `kappa`. Other parameters are passed through.
+
+#### `kappa.reset(name, cb)`
```

Comment:
I think it can be made hidden, yes. Instead advertize in the API to increase the view version number if you want to trigger a reindex. I used it in testing and benchmarking situations (eg to measure how long indexing takes).

Reply:
Sounds good. I agree with keeping it around undocumented for tests & benchmarks.